devito/builtins.py

```diff
@@ -123,46 +135,61 @@ def fset(f, g):
         else:
             raise NotImplementedError
 
-    lw = int(_order*sigma + 0.5)
+    try:
+        dtype = f.dtype.type
+    except AttributeError:
+        dtype = f.dtype
+
+    # TODO: Add s = 0 dim skip option
+    lw = tuple([int(_truncate*float(s) + 0.5) for s in as_tuple(sigma)])
+
+    if len(lw) == 1 and len(lw) < len(f.shape):
+        lw = len(f.shape)*(lw[0], )
+        sigma = len(f.shape)*(as_tuple(sigma)[0], )
+    elif len(lw) == len(f.shape):
+        sigma = as_tuple(sigma)
+    else:
+        raise ValueError("sigma must be an integer or a tuple of length" +
+                         " function.dimensions.")
 
     # Create the padded grid:
     objective_domain = ObjectiveDomain(lw)
+    # NOTE: This 'try' is required for distributed `f`.
```

Comment:
what happens if you use `shape_global` in both cases?

Reply:
Input can be a numpy array so won't have that property?

Reply:
ah, I see... perhaps you can add `shape = ...` **inside** the try-except that you already use for `dtype`

or at this point, you may well use `if isinstance(x, np.ndarray) ...`

anyway, nitpicking again, sorry